src/table/column-filter.scss

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+// See LICENSE.md in the project root for license terms and full copyright notice.
+@import '../style/index';
+@import '../menu/index';
+
+@mixin iui-column-filter {
+  @include iui-dropdown;
+  padding: $iui-baseline $iui-sm;
+  width: fit-content;
+
+  .iui-overlay {
```

Comment:
This classname seems too generic 😕

Reply:
I read your PR description, and it seems like `iui-input-combo` (which is not being used anywhere else) would make more sense here, and then you can get rid of `iui-overlay` completely.

Reply:
My thinking here was `.iui-input-combo` would, for example, place a text field + button side by side (plus some other styling like border radius and padding on the text field).
![Screen Shot 2021-04-28 at 3 15 17 PM](https://user-images.githubusercontent.com/849817/116460045-6e1e6080-a81b-11eb-9eb8-789db75855f4.png)

Then `.iui-input-combo` with the `.iui-overlay` modifier would allow you to instead place the button overtop of the text field instead of next to it.
![Screen Shot 2021-04-28 at 3 17 16 PM](https://user-images.githubusercontent.com/849817/116460239-b0e03880-a81b-11eb-9d71-2daaba13e2ba.png)

I have not thought far enough ahead for how to tackle something like this however. 🤔
![Screen Shot 2021-04-28 at 3 18 02 PM](https://user-images.githubusercontent.com/849817/116460316-c9e8e980-a81b-11eb-97f8-0f052ff11f10.png)

Reply:
If this was a borderless button: 
![image](https://user-images.githubusercontent.com/9084735/116460882-8f8c4600-a835-11eb-91c4-f6b270be7d6c.png)

Wouldn't that be the same thing as what you described with `iui-overlay`?

Reply:
This one's complicated to explain.  The borderless button overlaying the text field should only ever display an icon, no text.  I have a fixed padding so that the text field value doesn't clash with the borderless button.  You can see the padding in the image below.
![Screen Shot 2021-04-28 at 4 39 47 PM](https://user-images.githubusercontent.com/849817/116469610-361d1a80-a827-11eb-947b-64e02c8caf97.png)

If I swap the borderless button with the high-visibility and swap the icon with text, the text field's value will go underneath the button.
![Screen Shot 2021-04-28 at 4 43 55 PM](https://user-images.githubusercontent.com/849817/116470066-c9565000-a827-11eb-85b5-735a44e129de.png)

At first this seems ok, but you can not see the last few characters of whatever is typed into the text field.  The GIF below demos this well, the text inside is "This is the column filter in iTwinUI" but you can never see "UI" because the button covers it up.  I think this would be even worse if the button was in the front rather than at the rear.
![Screen Recording 2021-04-28 at 4 49 02 PM](https://user-images.githubusercontent.com/849817/116470771-b42df100-a828-11eb-9c1d-95a5369d19df.gif)

I thought about reversing it too, but if we place all buttons outside, then the borderless button will look out of place.
![Screen Shot 2021-04-28 at 4 54 52 PM](https://user-images.githubusercontent.com/849817/116471265-51892500-a829-11eb-9756-69340f84f6dc.png)

Reply:
Leaving this open because I still don't like `iui-icon-overlay` class 😄

Reply:
I don't like it, too. Because we won't be able to reuse Input component in React. We will need to enter whole input HTML just to wrap input with button in `iui-icon-overlay` class.

Reply:
We could probably handle the non-inline case in the future